plugins/README.adoc

```diff
@@ -7,3 +7,256 @@ NOTE: Work in Progress...
+
+Rationale::
+When hand-writing a complex argspec, the author may choose to build up to data structure from multiple dictionaries or vars.
+Other authors may choose to implement a complex, nested argspec as a single dictionary.
+Within a single collection, select one style and use it consistently.
+
+Examples::
+Single dictionary:
+https://github.com/ansible-collections/cisco.nxos/blob/3.0.0/plugins/module_utils/network/nxos/argspec/bgp_global/bgp_global.py
+
+Multiple dictionaries:
+https://github.com/ansible-collections/community.aws/blob/stable-3/plugins/modules/ec2_scaling_policy.py#L355-L370
+https://github.com/ansible-collections/amazon.cloud/blob/0.1.0/plugins/modules/backup_report_plan.py#L182-L234
+====
+
+
+== A plugin’s entry file should not contain more than 500 lines
+[%collapsible]
+====
+Explanations::
+A plugin’s entry file should not contain more than 500 lines
+
+Rationale::
+Long and complex code files can be difficult to maintain.
+Wherever possible, reusable functions and classes, such as those for data validation or manipulation, should be moved to module_utility or plugin_utility files and imported into plugins.
+This keeps the plugin code easier to read and maintain.
+500 lines is a general guideline that should serve most cases.
+====
+
+== Action plugins should be used when the code runs on the control node.
+[%collapsible]
+====
+Explanations::
+Use an action plugin for code that needs to run on the controller, where the host would be `localhost`.
+Rationale::
+Action plugins let you integrate local processing and local data with module functionality.
+Action plugins act in conjunction with modules to execute the actions required by playbook tasks.
+Using an action plugin will cause Ansible to skip creating a temporary tarball of execution code and runs the code directly on the controller host.
+
+Examples::
+See the `ansible.snmp.get` https://github.com/ansible-network/ansible.snmp/blob/main/plugins/action/get.py[action plugin] and associated https://github.com/ansible-network/ansible.snmp/blob/main/plugins/modules/get.py[module].
+
+====
+
+== Design decisions accounting for “jump hosts” should be avoided.
+[%collapsible]
+====
+Explanations::
+A plugin should be developed in accordance with Ansible best practices, without accounting for possible environment-specific variables.
+
+Rationale::
+The plugin author should prioritize development of the plugin's function.
+If an environment that the plugin could possibly run in (such as on a jump host) has some unique property this should be managed by the operator's Ansible configuration, separate from the plugin code.
+
+====
+
+== Plugins should be initially developed using the ansible plugin builder
+[%collapsible]
+====
+Explanations::
+The https://github.com/ansible-community/ansible.plugin_builder[ansible.plugin_builder] is a tool which helps developers scaffold new plugins.
+
+====
+
+== Pylint and type exceptions should have a specific exclusion type
```

Comment:
That's more generic python recommended practice, isn't it? Not wrong, but is it applicable to this document? Perhaps we should just point overall at some similar document of good practices, but for Python, and leave such topics out of here? This said, I don't know of such document, any idea welcome...